packages/nimble-components/src/table/specs/table-column-specs/table-column-formatted-text.md

```diff
@@ -161,22 +161,32 @@ Follows the [Column Type Philosophy](/packages/nimble-components/src/table/specs
+    -   `'nimble-unit-byte'` - Unit labels in this family are `byte`/`bytes`, `KB`, `MB`, `GB`, `TB`, `PB`
+    -   `'nimble-unit-byte-1024'` - Unit labels in this family are `byte`/`bytes`, `KiB`, `MiB`, `GiB`, `TiB`, `PiB`
 
 This column will display a blank cell when `typeof` the value is not `"number"` (i.e. if the value is `null`, `undefined`, not present, or has a different runtime data type). Note that IEE 754 numbers like Infinity, NaN, and -0 are type `"number"` so will be displayed how each formatter converts them. This will preserve values like `"∞"` and `"NaN"`.
 
 This column will trigger `invalidColumnConfiguration` on the table's validity state and will include flags in the column's validity state if its configuration can't be translated to a valid `Intl.NumberFormat` object. To provide better developer feedback about what's wrong with the configuration, the column could expose a public method like `createNumberFormat()` which would use the column's configuration to construct the formatter but allow any exceptions to be thrown.
 
+The `'human-readable'` format will rely on `Intl.NumberFormat` to translate unit strings (for the [units that it supports](https://tc39.es/ecma402/#table-sanctioned-single-unit-identifiers)) and localize the number (for comma/decimal). We will include our own logic for converting between unit values. For a family of units not supported by `Intl.NumberFormat`, we will provide our own translations (for some subset of languages) in a `nimble-unit-<name>` element. If the client requests a translation for one of these units in a language we don't support, we will fall back to English.
+
+Other libraries considered for number conversion and/or unit label localization include [globalizejs](https://github.com/globalizejs/globalize), [convert](https://convert.js.org/), [convert-units](https://github.com/convert-units/convert-units), and [iLib](https://github.com/iLib-js/iLib). None of these provided sufficient functionality to justify including them as a dependency.
+
+We will use `unit-family='nimble-unit-byte'` to display file sizes in SLE tables. Currently, SLE displays these values with the common `KB`/`MB`/`GB` unit labels, but uses a factor of 1024 to convert between units (which is not uncommon, but technically incorrect). Our formatting will use the same unit labels, but use a conversion factor of 1000. This will result in slightly different values being displayed. For SLE's internal consistency, we will also change their file size pipe to use a conversion factor of 1000 instead of 1024.
```

Comment:
To help clients maintain consistent formatting outside of the table, in the duration column we ended up deciding to expose a TS class with the formatting logic and an Angular pipe that wraps it. I think in your previous HLD you mentioned that might be low ROI but I'd like to revisit that in a quest for consistency with the duration column.

The challenge here is that there is potentially more configuration to specify when formatting a string. How would a client specify the unit family and any additional configuration to the format?

One idea is a static `format()` method on each unit element class which takes a locale and a configuration object. Nimble may not expose that via a pipe but clients could write their own pipe.

Thoughts?

Reply:
At first glance these ideas all feel a bit clunky to me (I actually hesitated to suggest it at all, hence the near-duplicate comment thread with this suggestion removed that I've now deleted). Either we duplicate a bunch of format configuration API or we require instantiating an element just to format a string. I don't want this question to hold up progress on the column work, so would be ok with leaving this as an open question and revisiting it when we start integrating into SLE.

Reply:
Ok, I moved the bit I added about creating an external formatter to the "Open Issues" section.